nupm/registry.nu

```diff
@@ -0,0 +1,178 @@
+export def --env add [
+    name: string,       # Name of the registry
+    url: string,        # URL or path to the registry
+    --save,             # Whether to commit the change to the registry index
+] {
+    if ($name in $env.NUPM_REGISTRIES) {
+        throw-error $"Registry '($name)' already exists. Use 'nupm registry update' to modify it."
+    }
+    $env.NUPM_REGISTRIES = $env.NUPM_REGISTRIES | insert $name $url
+
+    if $save {
+      $env.NUPM_REGISTRIES | save --force $env.NUPM_INDEX_PATH
+    }
+
+    print $"Registry '($name)' added successfully."
+}
+
+# Remove a registry
+@example "Remove a registry" { nupm registry remove my-registry }
+export def --env remove [
+    name: string        # Name of the registry to remove
+    --save,             # Whether to commit the change to the registry index
+] {
+    $env.NUPM_REGISTRIES = $env.NUPM_REGISTRIES | reject $name
+
+    if $save {
+      $env.NUPM_REGISTRIES | save --force $env.NUPM_INDEX_PATH
+    }
+
+    print $"Registry '($name)' removed successfully."
+}
+
+# Update a given registry url
+@example "Update registry URL" { nupm registry set-url my-registry https://new-url.com/registry.nuon }
+export def --env set-url [
+    name: string,   # Name of the registry to update
+    url: string,
+    --save,         # Whether to commit the change to the registry index
+]: nothing -> nothing {
+    $env.NUPM_REGISTRIES = $env.NUPM_REGISTRIES | update $name $url
+
+    if $save {
+      $env.NUPM_REGISTRIES | save --force $env.NUPM_INDEX_PATH
+    }
+
+    print $"Registry '($name)' URL updated successfully."
+}
+
+# https://www.nushell.sh/book/configuration.html#macos-keeping-usr-bin-open-as-open
+alias nu-rename = rename
+# Rename a registry
+@example "Rename a registry" { nupm registry rename my-registry our-registry }
+export def --env rename [
+    name: string,   # Name of the registry to update
+    new_name: string,
+    --save,         # Whether to commit the change to the registry index
+] {
+    $env.NUPM_REGISTRIES = $env.NUPM_REGISTRIES | nu-rename --column { $name: $new_name }
+
+    if $save {
+      $env.NUPM_REGISTRIES | save --force $env.NUPM_INDEX_PATH
+    }
+
+    print $"Registry '($name)' renamed successfully."
+}
+
+def init-index [] {
+    if not (nupm-home-prompt) {
+        throw-error "Cannot create NUPM_HOME directory."
+    }
+
+
+    if ($env.NUPM_INDEX_PATH | path exists) {
+        print $"Registry list already exists at ($env.NUPM_INDEX_PATH)"
+        return
+    }
+
+    $env.NUPM_REGISTRIES | save $env.NUPM_INDEX_PATH
+
+    print $"Registry index initialized at ($env.NUPM_INDEX_PATH)"
+}
+
+
+# Initialize a new nupm registry or a registry index if the `--index` flag is
+# passed in
+@example "Initialize registry index" { nupm registry init --index }
+export def init [--index] {
+    if $index {
+        init-index
+        return
+    }
+    # TODO initialize registry index here
```

Comment:
Here it should print something like "nupm registry init without --index is not implemented yet".